docs/guide/configuration.md

```diff
@@ -0,0 +1,104 @@
+# Configuration
+
+The Next Gen UI Agent can be configured in two ways: `programmatically` using Python dictionaries or `declaratively` using YAML configuration files.
+
+## Configuration Options
+
+### Core Settings
+
+- **`inference`** (`InferenceBase`): The LLM inference engine to use for component selection and configuration
```

Comment:
Should be better to for mat this as one chapter per option with description, allowed valuers etc